services/api-server/src/simcore_service_api_server/core/settings.py

```diff
@@ -152,17 +152,22 @@ def debug(self) -> bool:
         """If True, debug tracebacks should be returned on errors."""
         return self.SC_BOOT_MODE is not None and self.SC_BOOT_MODE.is_devel_mode()
 
-    @validator("API_SERVER_DEV_HTTP_CALLS_LOGS_PATH")
+    @validator("API_SERVER_DEV_HTTP_CALLS_LOGS_PATH", pre=True)
     @classmethod
     def _enable_only_in_devel_mode(cls, v, values):
```

Comment:
this is a **pre-validator** which means that `v` is directly what the caller is passing. Generally speaking it could be anything ...  but here you assume it is `str | None`. 

TIP1: add `v: Any` and the tooling will detect problems
TIP2: do not do everything in the pre-validator ... let the other validators do their job. E.g. do NOT return `Path(v)` since pydantic protocol will do that already (you annotated this field with `Path`)
TIP3: Carefule with `Path("")`
```cmd
>>> from pathlib import Path
>>> Path("")
PosixPath('.')
```

All together, my suggestion is


```python
@validator("API_SERVER_DEV_HTTP_CALLS_LOGS_PATH", pre=True)
@classmethod
def _enable_only_in_devel_mode(cls, v: Any, values: Mapping[str, Any]):
    if isinstance(v, str):
        path_str = v.strip()
        if not path_str:
            return None # otherwise it would Path("") which is converted to Path("."). Another possibility is to raise ValueError because we do not accept this value!

        if (values and (boot_mode := values.get("SC_BOOT_MODE")) and boot_mode.is_devel_mode() ):
            raise ValueError

    return v # Let now the other validators take care of this (e.g. check if it is not a string!)
```

Reply:
Thanks for the feedback. This looks good to me. I will "steal" that 😉